File: assets/stages/alley/stage.lua
function onCreateStage(stage)
    Game.defaultCamera.zoom = 0.8

    local sky = Sprite.new(-600, -200)
    sky:loadTexture("assets/stages/alley/sky.png")
    sky.scrollFactor.x = 00
    sky.scrollFactor.y = 0
    stage:add(sky)

    boyfriend.position.x = 1600
    boyfriend.position.y = 525
    stage:add(boyfriend)

    dad.position.x = 600
    dad.position.y = 525
    stage:add(dad)
end

File: src/funkin/objects/Stage.cpp
#include "Stage.hpp"

#include <fstream>
#include <iostream>

#include "funkin/modding/LuaScript.hpp"

namespace funkin::objects {
	Stage::Stage(const std::string &stageName, const std::shared_ptr<Character> &boyfriend, const std::shared_ptr<Character> &dad,
				 const std::shared_ptr<Character> &girlfriend) {

		this->stageName = stageName;
		if(!FileExists(("assets/stages/" + stageName + "/stage.lua").c_str())){
			TraceLog(5,("Unable to load stage '"+stageName+"'. Fallbacking to mainStage").c_str());
			this->stageName="mainStage";
		}
		script = std::make_shared<modding::LuaScript>("assets/stages/" + this->stageName + "/stage.lua");
		script->set("boyfriend", boyfriend);
		script->set("dad", dad);
		script->set("girlfriend", girlfriend);
		script->call("onCreateStage", this);
	}

	Stage::~Stage() = default;

	void Stage::snapToPosition(const std::shared_ptr<Character> &character, const float x, const float y) {
		character->position.x = x - character->hitbox.width / 2;
		character->position.y = y - character->hitbox.height;
	}
} // namespace funkin::objects
